cedar-policy/src/api.rs

```diff
@@ -627,80 +627,82 @@ impl Schema {
 /// Errors encountered during construction of a Validation Schema
 #[derive(Debug, Error)]
 pub enum SchemaError {
-    /// Errors loading and parsing schema files
-    #[error("JSON Schema file could not be parsed: {0}")]
-    ParseJson(serde_json::Error),
+    /// Error thrown by the `serde_json` crate during deserialization
+    #[error("failed to parse schema: {0}")]
+    Serde(#[from] serde_json::Error),
     /// Errors occurring while computing or enforcing transitive closure on
-    /// action id hierarchy.
-    #[error("Transitive closure error on action hierarchy: {0}")]
-    ActionTransitiveClosureError(String),
+    /// action hierarchy.
+    #[error("transitive closure computation/enforcement error on action hierarchy: {0}")]
+    ActionTransitiveClosure(String),
     /// Errors occurring while computing or enforcing transitive closure on
     /// entity type hierarchy.
-    #[error("Transitive closure error on entity hierarchy: {0}")]
-    EntityTransitiveClosureError(String),
-    /// Error generated when processing a schema file that uses features which
-    /// are not yet supported by the implementation.
-    #[error("Unsupported feature used in schema: {0}")]
-    UnsupportedSchemaFeature(String),
-    /// Undeclared entity type(s) used in an entity type's memberOf field or an
-    /// action's appliesTo fields.
-    #[error("Undeclared entity types: {0:?}")]
+    #[error("transitive closure computation/enforcement error on entity type hierarchy: {0}")]
+    EntityTypeTransitiveClosure(String),
+    /// Error generated when processing a schema file that uses unsupported features
+    #[error("unsupported feature used in schema: {0}")]
+    UnsupportedFeature(String),
```

Comment:
Instead of a `String` can we make this a pointer to the `UnsupportedFeature` enum in `cedar_validator_core`? (Unless we don't want to publicly expose that)

Reply:
Yes, I think the purpose of using string here is to avoid exposing too many Core structs/enums.